### ISSUE TITLE
Populate vTPM NVRAM data on boot

### ISSUE THREAD

**Issue by tylerfanelli** [stefano-garzarella/snp-svsm-vtpm#6]
Programs that use TPM services expect NVRAM data to be populated with things like EK certs, etc. Currently, a vTPM brought up with `remanufacture-vtpm.sh` doesn't include any of this. `swtpm` could be useful to bring up a vTPM.

@stefano-garzarella There seems to be a TODO in `remanufacture-tpm.sh` that talks about initializing the vTPM. Is this issue related to that?

**Comment by tylerfanelli** [comment]
Adding this issue to keep track of it. I'm currently working on importing swtpm into this project and using a script to initialize the TPM with it.

**Comment by stefano-garzarella** [comment]
> Programs that use TPM services expect NVRAM data to be populated with things like EK certs, etc. Currently, a vTPM brought up with `remanufacture-vtpm.sh` doesn't include any of this. 

In this PoC, if SVSM didn't find a signature in the state, automatically re-manufacture the vTPM.
So that script did a hack (documented in the script) just removing the SVSM state. So at the next boot, SVSM regenerated it.

>`swtpm` could be useful to bring up a vTPM.

SVSM uses MS-TPM, so I guess `swtpm` will store the state differently, I'm not sure if that can work.

> 
> @stefano-garzarella There seems to be a TODO in `remanufacture-tpm.sh` that talks about initializing the vTPM. Is this issue related to that?

Yep, the `ms-tpm-containerized-build` submodule in this repo was there for that. I used it in the first version, when the TPM state was injected by KBS, but for now it's not used because we are missing a tool that packages the vTPM state in a way that SVSM understands (create the FS, partition it, encrypt it with AES-XTS).

So, for now the easy way is to just start the CVM without any disk, in order to allow SVSM to re-manufacture the vTPM and create the state in the right way, so next boot can use it. I just pushed a commit which does exactly that: https://github.com/stefano-garzarella/snp-svsm-vtpm/commit/3baf3423b689daf1bd0e4810ac6d5b8124acc8a2
